### PR TITLE
Add foundational gun system

### PR DESCRIPTION
**Pull request by TheDrOfDoctoring**
`AbstractFirearm` handles the majority of logic for guns, including making use of the attribute system. All gun attributes are currently taken into account. Temporarily a `BasicFirearm` has been included as well as a default set of attributes for testing purposes. 
Bullets are added without a recipe for now and rendered.
Note block sound effects are currently used as placeholders for reloading sounds.

### CODE REVIEW

File: src/main/java/io/github/ageuxo/gloriousgunpowder/client/render/ClientEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package io.github.ageuxo.gloriousgunpowder.client.render;
+
+import io.github.ageuxo.gloriousgunpowder.event.GunFiredEvent;
+import io.github.ageuxo.gloriousgunpowder.item.AbstractFirearm;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.player.AbstractClientPlayer;
+import net.neoforged.bus.api.SubscribeEvent;
+import net.neoforged.neoforge.client.event.ClientTickEvent;
+
+public class ClientEventHandler {
```
Comment (ageuxo) [inline src/main/java/io/github/ageuxo/gloriousgunpowder/client/render/ClientEventHandler.java]: Maybe move this up one level to the client package, or we could make another client event handler there, and rename this one to RenderHandler or the like, I'll leave the choice up to you ;)

File: src/main/java/io/github/ageuxo/gloriousgunpowder/data/GunAttribute.java [hunk 2]
```diff
@@ -6,20 +6,27 @@
 import net.minecraft.network.codec.ByteBufCodecs;
 import net.minecraft.network.codec.StreamCodec;
 
-public record GunAttribute(float damage, float equipSpeed, float reloadSpeed) {
+public record GunAttribute(float damage, float accuracy, float reloadSpeed, float velocity, float recoil) {
     public static final Codec<GunAttribute> CODEC = RecordCodecBuilder.create(instance ->
             instance.group(
```
Comment (ageuxo) [inline src/main/java/io/github/ageuxo/gloriousgunpowder/data/GunAttribute.java:20]: Is this really the best way of storing the stats now that they're data defined? I don't know about the StreamCodec, but the regular Codec we could serialise it as a list of Objects mapping GunStat -> Value.
I could probably take a look at that doing that if you'd like.

File: src/main/resources/assets/glorious_gunpowder/sounds/cc/LICENSE.TXT [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Sound License
+The sounds in this directoy are available under different Creative Commons Licenses.
```
Comment (ageuxo) [inline src/main/resources/assets/glorious_gunpowder/sounds/cc/LICENSE.TXT]: directory*

**Review by ageuxo — CHANGES_REQUESTED** [review]
Just a few small things, otherwise looking very good!

File: src/main/java/io/github/ageuxo/gloriousgunpowder/data/GunAttribute.java (hunk 2 above)
Comment (TheDrOfDoctoring) [reply src/main/java/io/github/ageuxo/gloriousgunpowder/data/GunAttribute.java:20]: implemented temporary rework to make use of gunstats, may require future  changes

**Review by ageuxo — APPROVED** [review]
LGTM 👍

**Pull request merged**